generator/internal/descriptor_utils.cc

```diff
@@ -632,11 +632,29 @@ bool CheckParameterCommentSubstitutions() {
   return all_substitutions_used;
 }
 
+/// If a service name mapping exists, return the new name.
+/// Parses a command line argument in the form:
+/// {"service_name_mappings": "service_a:new_service_a,service:new_service"}.
+std::string GetEffectiveServiceName(VarsDictionary const& vars,
+                                    std::string const& name) {
+  auto service_name_mappings = vars.find("service_name_mappings");
+  if (service_name_mappings == vars.end()) return name;
+
+  std::vector<std::string> mappings =
+      absl::StrSplit(service_name_mappings->second, ',');
+  for (auto& mapping : mappings) {
+    std::vector<std::string> kv = absl::StrSplit(mapping, ':');
```

Comment:
You might as well use `absl::string_view` everywhere here, and avoid all the allocations/copying/deallocations.